blocks/components/editable/index.js

```diff
@@ -115,7 +133,16 @@ export default class Editable extends wp.element.Component {
 	componentDidUpdate( prevProps ) {
 		if ( this.props.tagName !== prevProps.tagName ) {
 			this.initialize();
-		} else if ( this.props.value !== prevProps.value ) {
+		}
+
+		if ( this.props.focus !== prevProps.focus && this.props.focus ) {
```

Comment:
@nylen proposition is valid for now because we don't store any offset yet. I updated in 7488f23.